### PR TITLE
*: add the description of `alter-primary-key`

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? 

Add the description of alter-primary-key and update the description or add/drop the primary key.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [x] This PR is to align with:pingcap/docs-cn#2199
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [x] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 1]
```diff
@@ -66,6 +66,12 @@ The TiDB configuration file supports more options than command-line parameters.
 - Determines whether to treat the `utf8` character set in old tables as `utf8mb4`.
 - Default value: `true`
 
+### `alter-primary-key`
+
+- Determine whether to add or remove the `primary key`.
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
- Determines whether to add or remove the primary key constraint to or from a column.
```
1. To unify the language styles in this document.
2. No need to wrap "primary key" with backticks.

Plus, I added some details to the sentence. Please confirm with the author about whether it's correct. @toutdesuite @zimulala

File: dev/reference/configuration/tidb-server/configuration-file.md [hunk 2]
```diff
@@ -66,6 +66,12 @@ The TiDB configuration file supports more options than command-line parameters.
...
+- Default value: `false`
+- Adding or dropping the `primart key` option is disabled by default. When `alter-primary-key` is set to `true`, the option is enabled. However, for the table that already existed, if the `primary key` is an integer, you can not drop the `primary key` of the table even if you set the variable to `true`.
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/configuration-file.md]:
```suggestion
- With this default setting, adding or removing the primary key constraint is not supported. You can enable this feature by setting `alter-primary-key` to `true`. However, if a table already exists before the switch is on, and the data type of its primary key column is an integer, dropping the primary key from the column is not possible even if you set this configuration item to `true`.
```

File: dev/reference/sql/statements/add-index.md [hunk 3]
```diff
@@ -68,7 +68,7 @@ mysql> EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 * `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
 * Descending indexes are not supported (similar to MySQL 5.7).
 * Adding multiple indexes at the same time is currently not supported.
-* It is not possible to add a `PRIMARY KEY` to a table.
+* Adding a `PRIMARY KEY` to a table is disabled by default, however, you can set the `alter-primary-key` to `true` to enable the feature. For more details, see [alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key).
```
Comment (yikeke) [inline dev/reference/sql/statements/add-index.md]:
```suggestion
* Adding the primary key constraint to a table is not supported by default. You can enable the feature by setting the `alter-primary-key` configuration item to `true`. For details, see [alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key).
```

File: dev/reference/sql/statements/create-index.md [hunk 4]
```diff
@@ -89,7 +89,7 @@ The global variables associated with the `CREATE INDEX` statement are `tidb_ddl_
 
 * `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
 * Descending indexes are not supported (similar to MySQL 5.7).
-* It is not possible to add a `PRIMARY KEY` to a table.
+* Adding a `PRIMARY KEY` to a table is not supported by default, however, you can set the `alter-primary-key` to `true` to enable the feature. For more details, see [alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key).
```
Comment (yikeke) [inline dev/reference/sql/statements/create-index.md]:
```suggestion
* Adding the primary key constraint to a table is not supported by default. You can enable the feature by setting the `alter-primary-key` configuration item to `true`. For details, see [alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key).
```

File: dev/reference/sql/statements/drop-index.md [hunk 5]
```diff
@@ -64,7 +64,7 @@ Query OK, 0 rows affected (0.30 sec)
 
 ## MySQL compatibility
 
-* Dropping the `PRIMARY KEY` is not supported.
+* Dropping the `PRIMARY KEY` is not supported by default, however, you can set the `alter-primary-key` to `true` to enable the feature. For more details, see [alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key).
```
Comment (yikeke) [inline dev/reference/sql/statements/drop-index.md]:
```suggestion
* Removing the primary key constraint from a column is not supported by default. You can enable the feature by setting the `alter-primary-key` configuration item to `true`. For details, see [alter-primary-key](/dev/reference/configuration/tidb-server/configuration-file.md#alter-primary-key).
```

**Comment by toutdesuite** [comment]
Docs on v3.0 and v3.1 have been updated. PTAL @zimulala

**Pull request merged**